preferences/keyboard.js

```diff
@@ -0,0 +1,114 @@
+    Logger.debug(settings.list_children())
+
+    const description = `${_('Syntax')}: &lt;Super&gt;h, &lt;Shift&gt;g, &lt;Shift&gt;&lt;Super&gt;h
+${_('Legend')}: &lt;Super&gt; - ${_('Windows  key')}, &lt;Primary&gt; - ${_('Control key')}
+${Msgs.prefs_keyboard_update_keys_instructions_text} <i>${Msgs.prefs_keyboard_update_keys_resets_label}</i> ${Msgs.prefs_keyboard_update_keys_resets_sub_1_label}`;
+
+    // TODO - calling this each time can introduce performance issues
+    // this.refSettings = this.buildRefSettings();
+
+    const group = new Adw.PreferencesGroup({ title: _('Update Shortcuts'), description });
+    for (const [prefix, gettextKey] of Object.entries({
+      window: 'Window Shortcuts',
+      workspace: 'Workspace Shortcuts',
+      con: 'Container Shortcuts',
+      focus: 'Focus Shortcuts',
+      prefs: 'Other Shortcuts',
+    })) {
+      const expander = new Adw.ExpanderRow({ title: _(gettextKey) });
+      KeyboardPage.createKeyList(settings, prefix).forEach(key =>
+        expander.add_row(new EntryRow({
+          title: key,
+          settings,
+          bind: key,
+          map: {
+            from(settings, bind) {
+              return settings.get_strv(bind).join(',');
+            },
+            to(settings, bind, value) {
+              const mappings = value.split(',').map(x => {
+                const [, key, mods] = Gtk.accelerator_parse(x);
+                return Gtk.accelerator_valid(key, mods) && Gtk.accelerator_name(key, mods);
+              });
+              if (mappings.every(x => !!x)) {
+                Logger.info('setting', bind, 'to', mappings);
+                settings.set_strv(bind, mappings)
+              }
+            }
+          }
+        })))
+      group.add(expander);
+    }
+
+    this.add(group);
+
+    const modmapGroup = new Adw.PreferencesGroup({
+      title: Msgs.prefs_keyboard_other_mod_mask_header,
+      description: `<i>${Msgs.prefs_keyboard_other_mod_mask_informational1}</i> ${Msgs.prefs_keyboard_other_mod_mask_informational2}`
+    });
+          modmapGroup.add(new RadioRow({
+            title: Msgs.prefs_keyboard_mod_mask_tile_label,
+            settings,
+            bind: 'mod-mask-mouse-tile',
+            options: {
+              Super: _('Super'),
+              Ctrl: _('Ctrl'),
+              Alt: _('Alt'),
+              None: _('None'),
+            }
+          }));
+    this.add(modmapGroup);
+  }
+
+  static createKeyList(settings, categoryName) {
+    return settings
+      .list_keys()
+      .filter((keyName) => !!keyName && !!categoryName && keyName.startsWith(categoryName))
+      .sort((a, b) => {
+          const aUp = a.toUpperCase();
+          const bUp = b.toUpperCase();
+          if (aUp < bUp) return -1;
+          if (aUp > bUp) return 1;
+          return 0;
+        });
+  }
+
+  // TODO move this to keybindings.js or settings.js
+  buildRefSettings() {
```

Comment:
i noticed this never gets called

Reply:
This function was intended for querying the system for existing keybindings, annotate the keybinding list with conflicts and then show it on the KB page. But did not get time to implement it.

Reply:
To resolve this, let's leave it be.